packages/webdriverio/src/index.ts

```diff
@@ -21,37 +12,53 @@ import {
   axeRunLegacy
 } from './utils';
 
+import type { Browser } from 'webdriverio';
+import type { RunOptions, AxeResults, ContextObject } from 'axe-core';
+import type {
+  Options,
+  CallbackFunction,
+  WdioBrowser,
+  WdioElement,
+  PartialResults
+} from './types';
+
 export default class AxeBuilder {
-  private client: BrowserObject;
+  private client: Browser<'async'>;
   private axeSource: string;
-  private includes: string[];
-  private excludes: string[];
-  private option: RunOptions;
-  private disableFrameSelectors: string[];
+  private includes: string[] = [];
+  private excludes: string[] = [];
+  private option: RunOptions = {};
+  private disableFrameSelectors: string[] = [];
   private legacyMode = false;
 
   constructor({ client, axeSource }: Options) {
     assert(
       isWebdriverClient(client),
       'An instantiated WebdriverIO client greater than v5 is required'
     );
-    const sourceDir = require.resolve('axe-core');
-    const source = fs.readFileSync(sourceDir, 'utf-8');
-    this.client = client;
-    this.axeSource = axeSource || source;
-    this.includes = [];
-    this.excludes = [];
-    this.option = {};
-    this.disableFrameSelectors = [];
+    // Treat everything as Browser<'async'>:
+    // - Anything sync can also run async, since JS can await sync functions
+    // - Ignore MultiRemoteBrowser, which is just Browser with extra props
+    this.client = client as Browser<'async'>;
```

Comment:
Needed to simplify this a little. Typescript does not like the union type `.execute()` here.